pkgs/shells/rc/default.nix

```diff
@@ -1,42 +1,42 @@
-{ lib, stdenv, fetchurl, autoreconfHook
-, ncurses #acinclude.m4 wants headers for tgetent().
-, historySupport ? false
-, readline ? null
-}:
+{ lib, stdenv, fetchFromGitHub, autoreconfHook, byacc
+, ncurses # acinclude.m4 wants headers for tgetent().
+, historySupport ? false, readline ? null }:
 
 stdenv.mkDerivation rec {
   pname = "rc";
-  version = "1.7.4";
+  version = "unstable-2021-08-03";
 
-  src = fetchurl {
-    url = "http://static.tobold.org/rc/rc-${version}.tar.gz";
-    sha256 = "1n5zz6d6z4z6s3fwa0pscqqawy561k4xfnmi91i626hcvls67ljy";
+  src = fetchFromGitHub {
+    owner = "rakitzis";
+    repo = "rc";
+    rev = "8ca9ab1305c3e30cd064290081d6e5a1fa841d26";
+    sha256 = "0744ars6y9zzsjr9xazms91qy6bi7msg2gg87526waziahfh4s4z";
   };
 
   nativeBuildInputs = [ autoreconfHook ];
-  buildInputs = [ ncurses ]
+  buildInputs = [ byacc ncurses ]
     ++ lib.optionals (readline != null) [ readline ];
 
   configureFlags = [
     "--enable-def-interp=${stdenv.shell}" #183
     ] ++ lib.optionals historySupport [ "--with-history" ]
     ++ lib.optionals (readline != null) [ "--with-edit=readline" ];
 
+  #reproducible-build
   prePatch = ''
     substituteInPlace configure.ac \
-      --replace "date -I" "echo 2015-05-13" #reproducible-build
+      --replace "$(git describe || echo '(git description unavailable)')" "${builtins.substring 0 7 src.rev}"
```

Comment:
Git is not available and fetchFromGitHub doesn't download the full repo by default

Reply:
If we really need this use ´´fetchFromGitHub.rev``.

Reply:
> Git is not available and fetchFromGitHub doesn't download the full repo by default
@Mindavi That was my intention. The ``git describe`` call is used exclusively for setting the ``$version`` environment variable. e.g.
1. execute ``rc``.
2. in rc's shell, run ``echo $version``.

``src.rev`` returns the same hash so I just trimmed it a bit to spare us having to do a full git clone.

> If we really need this use ´´fetchFromGitHub.rev``.

@SuperSandro2000 I'm not sure I'm following. Is something wrong with ``src.rev``?

Reply:
Should be fine like this.